### DISCUSSION TITLE
Federation "Light"

### DISCUSSION THREAD

**Discussion by t1** [smallrye/smallrye-graphql#902] [Ideas]
[GraphQL Federation](https://www.apollographql.com/docs/federation/federation-spec/) is probably the most important reason to choose GraphQL over REST. I was wondering if it could be simplified even further.

1. Every `ID` type field is by default equivalent to the containing type having a directive `@key(fields = <fieldName>)`. This would allow many existing types to be extendable by other services.
2. The `@extends` and `@external` directives are optional. For some types there is no clear 'main' type, e.g. within the different bounded contexts of an `Order`, e.g. billing and fulfillment, none is clearly the 'core'. Only the type name and the id field may be common to all services and introducing a hierarchy may seem inappropriate; all services would be equally just extensions.

Maybe I'll have to write such a federation gateway to prove that it's possible. What do you think?

**Comment by phillip-kruger** [comment]
Interesting, it might be something we enable by config ? I think it can work, but I also thing we need to focus on the proper solution. Maybe me and you can meet sometime like we discussed and see if we can not finalize what you have started ?

**Reply by t1** [reply]
The first part could be done also in the backend service and should have to explicitly be enabled with a config.

But my idea was more for a new federation gateway. It could work without any directives by default.
